data-prepper-plugins/s3-sink/src/integrationTest/java/org/opensearch/dataprepper/plugins/sink/S3ObjectWorkerIT.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.sink;
+
+import static org.junit.Assert.assertNotNull;
+import static org.mockito.Mockito.atLeastOnce;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import java.time.Duration;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.LinkedHashMap;
+import java.util.LinkedList;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.BlockingQueue;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.opensearch.dataprepper.model.configuration.PluginModel;
+import org.opensearch.dataprepper.model.event.Event;
+import org.opensearch.dataprepper.model.event.JacksonEvent;
+import org.opensearch.dataprepper.model.log.JacksonLog;
+import org.opensearch.dataprepper.model.record.Record;
+import org.opensearch.dataprepper.model.types.ByteCount;
+import org.opensearch.dataprepper.plugins.sink.accumulator.BufferTypeOptions;
+import org.opensearch.dataprepper.plugins.sink.configuration.BucketOptions;
+import org.opensearch.dataprepper.plugins.sink.configuration.ObjectKeyOptions;
+import org.opensearch.dataprepper.plugins.sink.configuration.ThresholdOptions;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.s3.S3Client;
+
+class S3ObjectWorkerIT {
+    private S3Client s3Client;
+    private final BlockingQueue<Event> eventQueue;
+    private S3SinkConfig s3SinkConfig;
+    private S3SinkService s3SinkService;
+    private static final int EVENT_QUEUE_SIZE = 100000;
+    private String codecFileExtension = null;
+    private static final String DEFAULT_CODEC_FILE_EXTENSION = "json";
+
+    public S3ObjectWorkerIT() {
+        eventQueue = new ArrayBlockingQueue<>(EVENT_QUEUE_SIZE);
+    }
+
+    @BeforeEach
+    public void setUp() {
+
+        s3Client = S3Client.builder().region(Region.of(System.getProperty("tests.s3source.region"))).build();
+        String bucket = System.getProperty("tests.s3source.bucket");
+
+        s3SinkConfig = mock(S3SinkConfig.class);

```

Comment:
This is an integration test and should not be using mocks. Please create real objects.